java/flight/src/main/java/org/apache/arrow/flight/FlightProducer.java

```diff
@@ -181,5 +183,11 @@ void doAction(CallContext context, Action action,
 
     /** Whether the call has been cancelled by the client. */
     boolean isCancelled();
+
+    /** Get the middleware instance of the given type for this call. */
```

Comment:
might be worth clarifying if this returns null or throws an exception if the middle was isn't found?